### DISCUSSION TITLE
Implement a Package Transaction History

### DISCUSSION THREAD

**Discussion by NICHOLAS85** [wstyres/Zebra#740] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'd like the idea of being able to see a log of all apt transactions within Zebra at least to a certain point in order to keep track of what tweaks were removed/installed and when.

**Describe the solution you'd like**
Including a transaction history menu option somewhere which then opens to a view with a log, stylized or not, could be a solution.

**Describe alternatives you've considered**
Simply keeping track myself or viewing the logs in filza, although this log file is not very pretty (obviously) and includes some information which can make it quite verbose.

Some additional features which could make the transaction history view more meaty/useful can be performing actions from this view, ie reinstalling a removed tweak or undoing a transaction which removed and installed multiple tweaks. Another idea is adding an option to write notes for specific transactions to keep track of why a certain package was removed.

**Comment by wstyres** [comment]
So basically something like a log containing each action you performed as an option?

**Comment by NICHOLAS85** [comment]
Yea, something which at a glance lets you see a log but with interaction can perform actions on those items. ie via swipe gestures or by tapping on the transaction for a screen with options. However you think it would best work!

**Comment by wstyres** [comment]
Ok, we can look into this for a future version. User Interface wise might be a bit difficult and it might be taxing for Zebra to perform these calculations but we will see how it goes when we start implementing it.

**Comment by kevini15** [comment]
<img width="663" alt="Bildschirmfoto 2020-02-18 um 21 25 33" src="https://user-images.githubusercontent.com/17512423/74775086-4043a080-5295-11ea-8825-4483e41a2dfb.png">

This is how I currently manage my history - it's quite painful and time consuming, but totally worth it. Makes it much easier to narrow down which recently installed tweak is causing issues. This spreadsheet however doesn't factor in recently updated tweaks - only installs and removals.

Would love to see this feature in Zebra!
